data_structures/src/utxo_pool.rs

```diff
@@ -147,6 +150,51 @@ impl OwnUnspentOutputsPool {
     }
 }
 
+/// Struct that keeps the unspent outputs pool and the own unspent outputs pool
+#[derive(Debug)]
+pub struct NodeUtxos<'a> {
+    pub all_utxos: &'a UnspentOutputsPool,
+    pub own_utxos: &'a mut OwnUnspentOutputsPool,
+}
+
+impl<'a> UtxoFantasy for NodeUtxos<'a> {
+    fn sort_iter(&self, strategy: UtxoSelectionStrategy) -> Vec<OutputPointer> {
+        match strategy {
+            UtxoSelectionStrategy::BigFirst => self.own_utxos.sort(&self.all_utxos, true),
+            UtxoSelectionStrategy::SmallFirst => self.own_utxos.sort(&self.all_utxos, false),
+            UtxoSelectionStrategy::Random => {
+                self.own_utxos.iter().map(|(o, _ts)| o.clone()).collect()
+            }
+        }
+    }
+
+    fn get_time_lock(&self, outptr: &OutputPointer) -> Option<u64> {
+        let time_lock = self.all_utxos.get(outptr).map(|vto| vto.time_lock);
+        let time_lock_by_used = self.own_utxos.get(outptr).copied();
+
+        match (time_lock, time_lock_by_used) {
+            (Some(a), Some(b)) => Some(std::cmp::max(a, b)),
+            (Some(a), None) => Some(a),
+            (None, Some(b)) => Some(b),
+            _ => None,
+        }
+    }
+
+    fn get_value(&self, outptr: &OutputPointer) -> Option<u64> {
+        self.all_utxos.get(outptr).map(|vto| vto.value)
+    }
+
+    fn get_included_block_number(&self, outptr: &OutputPointer) -> Option<u32> {
+        self.all_utxos.included_in_block_number(outptr)
+    }
+
+    fn set_used_output_pointer(&mut self, outptr: &OutputPointer, ts: u64) {
+        if let Some(current_ts) = self.own_utxos.get_mut(outptr) {
```

Comment:
I don't like that this can silently fail. Can we use an unwrap here? Or is it better to return an error and let the called handle it?

Reply:
the function set_used_output_pointer is called when the utxos were previously selected so it should not fail. I wouldnt like too much add an Error, to keep the function as simple as possible

Reply:
Then I prefer using an unwrap, we know that this cannot fail so if it fails it means there is an error in our code.